lua/plenary/path.lua

```diff
@@ -277,6 +279,51 @@ function Path:read()
   return data
 end
 
+local read_head_or_tail = function(file, lines, start, concat, in_ch)
+  local fd = assert(uv.fs_open(file, "r", 438))
+  local stat = assert(uv.fs_fstat(fd))
+
+  local data = ''
+  local index = start(stat)
+  local count = 0
+
+  while count < lines do -- or end of file. DO CHECK HERE
+    local char = assert(uv.fs_read(fd, 1, index))
+    if char == '\n' then
+      count = count + 1
+      if count < lines then data = concat(data, char) end
+    else
+      data = concat(data, char)
+    end
+    index = in_ch(index)
+  end
+  assert(uv.fs_close(fd))
+
+  return data
+end
+
+function Path:head(lines)
+  lines = lines or 10
+  self = check_self(self)
+
+  return read_head_or_tail(self:expand(), lines, function()
+    return 0
+  end, function(data, char)
+    return data .. char
+  end, function(index) return index + 1 end)
+end
+
+function Path:tail(lines)
+  lines = lines or 10
+  self = check_self(self)
+
+  return read_head_or_tail(self:expand(), lines, function(stat)
+    return stat.size - 2
+  end, function(data, char)
+    return char .. data
+  end, function(index) return index - 1 end)
+end
+
```

Comment:
Not really sure how to read `head`/`tail` correctly with `fs`.
Works similar to `unix` `head`/`tail` if `lines == nil` then we will read 10 lines. Otherwise do the other thing.
We should also stop if we read the whole file. Will add that as well later.